packages/block-library/src/button/style.scss

```diff
@@ -37,7 +37,7 @@ $blocks-button__height: 56px;
 }
 
 .wp-gs .wp-block-button__link:not(.has-background) {
-	background-color: var(--wp-block-core-button--color--background, var(--wp-color--primary, $dark-gray-700));
+	background-color: var(--wp--color--primary);
```

Comment:
Should we pass $dark-gray-700 as a fallback?

Reply:
Note that `--wp--color--primary` is going to always have a value (user-defined, the theme default, or core default), so the fallback won't be necessary.